contracts/DecentHats_0_1_0.sol

```diff
@@ -195,6 +195,29 @@ contract DecentHats_0_1_0 {
         }
     }
 
+    function createRoleHat(
+        IHats hatsProtocol,
+        uint256 adminHatId,
+        Hat calldata hat,
+        uint256 topHatId,
+        address topHatAccount,
+        IERC6551Registry registry,
+        address hatsAccountImplementation,
+        bytes32 salt
+    ) public returns (uint256 hatId, address accountAddress) {
+        (hatId, accountAddress) = createHatAndAccountAndMintAndStreams(
+            hatsProtocol,
+            adminHatId,
+            hat,
+            topHatAccount,
+            registry,
+            hatsAccountImplementation,
+            salt
+        );
+
+        hatsProtocol.transferHat(topHatId, address(this), msg.sender);
```

Comment:
Can you please leave some natspec comments on this function, explaining how it's independent of the rest of the code in this contract, and why it exists, and the reasoning behind its implementation?